File: game-logic-core/src/main/scala/fwc/gameLoading/WildlingCard.scala
package fwc.gameLoading

import fwc.JsonSerializable
import ujson.Value

case class WildlingCard(
                         code: Int,
                         title: String,
                         wildlingVictoryLowestBidderText: String,
                         wildlingVictoryEveryoneElseText: String,
                         playerVictoryText: String
                       ) extends CardTrait with JsonSerializable{
  override def getCode: Int = code

  override def toJson: Value = ujson.Obj(
    "code" -> code,
    "title" -> title,
    "wildlingVictoryLowestBidderText" -> wildlingVictoryEveryoneElseText,
    "wildlingVictoryEveryoneElseText" -> wildlingVictoryEveryoneElseText,
    "playerVictoryText" -> playerVictoryText,
  )
}

File: bots/server_module/game_rules/wildling_card.py
class WildlingCard:
    def __init__(
            self,
            code: int,
            title: str,
            wildling_victory_lowest_bidder_text: str,
            wildling_victory_everyone_else_text: str,
            player_victory_text: str,
    ):
        self.code = code
        self.title = title
        self.wildling_victory_lowest_bidder_text = wildling_victory_lowest_bidder_text
        self.wildling_victory_everyone_else_text = wildling_victory_everyone_else_text
        self.player_victory_text = player_victory_text

    @classmethod
    def from_json(cls, json):
        return cls(
            json['code'],
            json['title'],
            json['wildlingVictoryLowestBidderText'],
            json['wildlingVictoryEveryoneElseText'],
            json['playerVictoryText']
        )
